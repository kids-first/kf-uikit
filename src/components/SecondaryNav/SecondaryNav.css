.SecondaryNav {
    @apply
      w-full
      h-64
      bg-white;
}

.SecondaryNav--navigation {
    @apply
      relative
      bg-white
      overflow-hidden
      border-mediumGrey
      border;
}

.SecondaryNav--navigation ul {
    @apply
      p-0
      m-0
      relative
      inline-block
      h-full
      align-middle;
}

.SecondaryNav--link {
    @apply
      p-0
      m-0
      relative
      inline-block
      h-full
      border-mediumGrey
      border-r
      align-middle;

    a {
        @apply
          block
          border-transparent
          border-b-2
          h-full
          w-full
          px-5
          py-4
          text-center
          font-title
          font-semibold
          no-underline
          text-sm
          text-purple;
    }

    a:hover {
        @apply
          text-purpleHover;
    }
}

.SecondaryNav--link-active {
    a {
        @apply border-purpleHover;
    }
}
