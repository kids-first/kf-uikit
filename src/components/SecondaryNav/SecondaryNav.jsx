import React from 'react';
import propTypes from 'prop-types';
import className from 'classnames';
import './SecondaryNav.css';


/**
 * A navigation item to be used in the secondary nav bar
 *
 */
export const NavLink = ({ active, children }) => {

  const linkClass = className({
    'SecondaryNav--link': true,
    'SecondaryNav--link-active': active,
  });

  return (
    <li className={linkClass}>{children}</li>
  );
}

NavLink.propTypes = {
  /** Child elements to display on the navbar */
  children: propTypes.node,
  /** Whether the link is active */
  active: propTypes.bool,
};

NavLink.defaultProps = {
  children: null,
  active: false,
};


/**
 * Secondary navigation typically used under the header
 *
 * Must pass an array of NavLinks for the navigation bar.
 * Each NavLink should contain an anchor with appropriate href, or a react
 * router Link or equivelant.
 */
const SecondaryNav = ({ buttons }) => {

  const navClass = className(
    'SecondaryNav--navigation',
  );

  return (
    <nav className={navClass}>
      <ul>
        {buttons ? buttons.map(button => button) : null}
      </ul>
    </nav>
  );
}

SecondaryNav.propTypes = {
  /** NavLinks to display on the navbar */
  buttons: propTypes.node,
};

SecondaryNav.defaultProps = {
  buttons: null,
};

export default SecondaryNav;
