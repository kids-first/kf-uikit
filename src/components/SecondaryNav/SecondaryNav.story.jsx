/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import SecondaryNav, { NavLink } from './SecondaryNav';

const stories = storiesOf('SecondaryNav', module);

stories.addDecorator(withKnobs);

stories.add('SecondaryNav', () => (
  <SecondaryNav
    buttons={[
      <NavLink active key={1}><a href="/">Dashboard</a></NavLink>,
      <NavLink key={2}><a href="/">File Repository</a></NavLink>]}
  />
));
