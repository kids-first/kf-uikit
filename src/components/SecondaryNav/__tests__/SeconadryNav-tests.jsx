import React from 'react';
import renderer from 'react-test-renderer';
import SecondaryNav, { NavLink } from '../SecondaryNav';

it(`secondary navigation renders correctly`, () => {
    const tree = renderer.create(
      <SecondaryNav
        buttons={[<NavLink key={1}><a href="/">Testing</a></NavLink>]}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
