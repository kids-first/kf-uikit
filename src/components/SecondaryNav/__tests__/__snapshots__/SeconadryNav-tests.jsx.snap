// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`secondary navigation renders correctly 1`] = `
<nav
  className="SecondaryNav--navigation"
>
  <ul>
    <li
      className="SecondaryNav--link"
    >
      <a
        href="/"
      >
        Testing
      </a>
    </li>
  </ul>
</nav>
`;
